### PR TITLE
fix: prop types for styled element: JSX.IntrinsicElements[T]

### PR DESCRIPTION
**Pull request by btakita**

### CODE REVIEW

**Comment by btakita** [comment]
https://github.com/solidjs/solid-styled-components/issues/2

File: test/test.spec.tsx [hunk 1]
```diff
@@ -29,6 +29,23 @@ describe("Simple Styled", () => {
     });
   });
 
+  test("Creates input properly", () => {
+    const Input = styled('input')`
```
Comment (dragos199993) [inline test/test.spec.tsx:33]: I might be wrong, but the issue was pointing out that we should handle the case of something like:
```
const SpecialInput = styled(Input)``
```
Isn't worth to add a test case where you test this too?

**Comment by thdxr** [comment]
This is the right fix let's merge it!

**Comment by ryansolid** [comment]
Thank you.

**Pull request merged**